src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -110,47 +133,333 @@ class boss_gluth : public CreatureScript
+        }
+    };
+
+    SpellScript* GetSpellScript() const
+    {
+        return new spell_gluth_zombiechow_search_SpellScript();
+    }
+
+};
+
+// creature 16360 (10man) / 30303 (25man)
+class npc_zombie_chow : public CreatureScript
+{
+public:
+
+    npc_zombie_chow() : CreatureScript("npc_zombie_chow") { }
+
+    struct npc_zombie_chowAI : public ScriptedAI
+    {
+        npc_zombie_chowAI(Creature* creature) : ScriptedAI(creature)
+        {
+            GluthGUID = creature->GetInstanceScript()->GetGuidData(DATA_GLUTH);
+
+            DoCast(me, SPELL_INFECTED_WOUND);
+            state = STATE_ZOMBIE_NORMAL;
+        }
+
+        void UpdateAI(uint32 /*diff*/) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            Creature* gluth = ObjectAccessor::GetCreature(*me, GluthGUID);
+            if (state == STATE_ZOMBIE_DECIMATED)
+            {
+                // Putting this in the UpdateAI loop fixes an issue where death gripping a decimated zombie would make the zombie stand still until the rest of the fight.
+                // Also fix the issue where if one or more zombie is rooted when decimates hits (and MovePoint() is called), the zombie teleport to the boss. pretty weird behavior.
+                if (me->IsStopped() && me->CanFreeMove() && me->GetExactDist2d(gluth) > 10.0)
+                {
+                    if (Creature* gluth = ObjectAccessor::GetCreature(*me, GluthGUID))
+                        me->GetMotionMaster()->MovePoint(0, gluth->GetPosition()); // isnt dynamic. point to point move without taking into account that Gluth might change his position before reaching destination point.
+                        //me->GetMotionMaster()->MoveFollow(gluth, 0, 0); // the zombies will stand still too far away from Gluth at the end of their movement. Also doesnt seem to work with SetWalk(true)
+                        //me->GetMotionMaster()->MoveChase(gluth); // crash server. I assume its because Gluth is friendly. We need the zombies to get really close to Gluth's center position anyway (10 yards) for the multi kill spell to work.
+
+                    // Using MovePoint() works but creates a small bug: if Gluth is in a certain position when a zombie starts its movement
+                    // toward him and Gluth moves, by the time the zombie reach Gluth's initial position, Gluth might have moved meanwhile. 
+                    // As soon as players will notice this bug, they might adapt their strategy which in turn will change the nature of the fight.
+                }
+
+                if (me->GetExactDist2d(gluth) <= 10.0)
+                    me->StopMoving();
+            }
+            else if (state == STATE_ZOMBIE_NORMAL)
                 DoMeleeAttackIfReady();
         }
+
+        void SetData(uint32 id, uint32 value) override
+        {
+            if (id == 1) // change of state
```

Comment:
i'm gonna remove the second parameter and rename the first parameter as "value" since there is only one use of SetData. Should be clearner. I want to avoid creating an Enum for just on element.